app/models/spree/referral_setting.rb

```diff
@@ -0,0 +1,4 @@
+module Spree
+  class ReferralSetting < Spree::Base   
```

Comment:
This shouldn't be inherited from Spree::Base. It should inherit from Spree::Preference